### PR TITLE
(Update) PHP8 Support

### PR DESCRIPTION
**Pull request by HDVinnie**

### CODE REVIEW

**Comment by HDVinnie** [comment]
@owenvoke @Elhebert anyway this can be merged?

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
         }
     ],
     "require": {
-        "php": "^7.2.5",
+        "php": "^7.2 | ^7.3 | ^7.4 | ^8.0",
```
Comment (owenvoke) [inline composer.json]: This can be simplified to the following to support 7.3 and 7.4. :+1:
```suggestion
        "php": "^7.2.5 | ^8.0",
```

**Comment by owenvoke** [comment]
Nice, I don't actually have any say in this package (I'm just a past contributor). But probably seems fine. Might be worth updating the CI workflows as well to run against PHP 8. 👍🏻 If you need any help with that, let me know.

**Comment by HDVinnie** [comment]
@owenvoke thanks. Updated the PHPUnit configs and the composer.json with your suggestion.

**Comment by Elhebert** [comment]
Thanks for the PR, the code looks good. However some tests are failing due to some deprecation. Can you have a look?

File: .github/workflows/phpunit-l7.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                php: [7.4, 7.3, 7.2.5]
+                php: [8.0snapshot, 7.4, 7.3, 7.2.5]
```
Comment (Elhebert) [inline .github/workflows/phpunit-l7.yml]: Any reason for using `8.0snapshot` over `8.0`?

**Comment by owenvoke** [comment]
Looks like the depreciations are due to versions of Mockery lower than 1.4.1. 🤔

When running on the `--prefer-lowest` on PHP 8, it seems to select the exact version that is specified by `orchestra/testbench-core`, which for Testbench 4.5 is Mockery 1.2.3, not sure how to force that to be the latest version without adding Mockery `^1.4.2` to the Composer JSON.

File: .github/workflows/phpunit-l7.yml (hunk 2 above)
Comment (HDVinnie) [reply .github/workflows/phpunit-l7.yml]: My bad. I thought you were using travisci which only has snapshot as of now. Will update.

**Comment by HDVinnie** [comment]
> Looks like the depreciations are due to versions of Mockery lower than 1.4.1. 🤔
> 
> When running on the `--prefer-lowest` on PHP 8, it seems to select the exact version that is specified by `orchestra/testbench-core`, which for Testbench 4.5 is Mockery 1.2.3, not sure how to force that to be the latest version without adding Mockery `^1.4.2` to the Composer JSON.

I have update tests and composer.json but IMO for them to all pass will need to drop PHP 7.2 support and only allow 7.3, 7.4 and 8.0.

**Comment by HDVinnie** [comment]
@Elhebert all tests are now passing. Would mean the new release oof this package would require PHP 7.3+

**Comment by owenvoke** [comment]
I feel like that's probably fine. PHP 7.2 is [EOL](https://www.php.net/eol.php), and people can still use the existing version perfectly fine if they are on 7.2. 🤔

**Comment by Elhebert** [comment]
Do we still need to `mockery` as a dev dependency if we drop 7.2?

**Comment by HDVinnie** [comment]
> Do we still need to `mockery` as a dev dependency if we drop 7.2?

Yes for PHP8. New release of this package should be v3.0.0 and requires PHP 7.3+

**Comment by Elhebert** [comment]
I already started work on a v3 (#57) which drop the support of Laravel 7 (and PHP 7.2.5).

I'm going to merge (soon) #62 and #65 into master. Once this is done, you can rebase and see if we still need to have `mockery` as a dev dependency or not.

I'll try to do it before the end of the day (in the next 6-8 hours).

That way we can release v3 with PHP8 support.

**Comment by owenvoke** [comment]
If it's Laravel 8+, that should be fine. 👍🏻

**Comment by Elhebert** [comment]
@HDVinnie just merged both branches. You can now rebase on master ;)

**Comment by HDVinnie** [comment]
@Elhebert PR is good to go.

**Comment by owenvoke** [comment]
Should be able to just do `^7.3 || ^8.0` for the PHP version. 👍🏻 Otherwise this looks great.

File: composer.json [hunk 3]
```diff
@@ -13,12 +13,13 @@
         }
     ],
     "require": {
-        "php": "^7.3",
+        "php": "^7.3 | ^7.4 | ^8.0",
```
Comment (Elhebert) [inline composer.json:16]:
```suggestion
        "php": "^7.3 || ^8.0",
```

**Review by Elhebert — APPROVED** [review]

File: composer.json (hunk 3 above)
Comment (HDVinnie) [reply composer.json:16]: This does the same thing.....just comes down to personal preference. I always like clearly defining each version. If you dont then ill change it.

**Comment by Elhebert** [comment]
[v3](https://github.com/Elhebert/laravel-sri/releases/tag/3.0.0) has been tagged and released 🎉

**Pull request merged**